modin/engines/base/frame/data.py

```diff
@@ -1732,6 +1732,39 @@ def from_pandas(cls, df):
             dtypes=new_dtypes,
         )
 
+    @classmethod
+    def from_arrow(cls, at):
+        """Improve simple Arrow Table to an advanced and superior Modin DataFrame.
+
+        Args:
+            at: Arrow Table object.
+
+        Returns:
+            A new dataframe.
+        """
+        new_frame, new_lengths, new_widths = cls._frame_mgr_cls.from_arrow(at, True)
+        new_columns = Index.__new__(Index, data=at.column_names, dtype="O")
+        new_index = Index.__new__(RangeIndex, data=range(at.num_rows))
+        new_dtypes = pandas.Series(
+            [cls._arrow_type_to_dtype(i.type) for i in at.columns],
+            index=at.column_names,
+        )
+        return cls(
+            partitions=new_frame,
+            index=new_index,
+            columns=new_columns,
+            row_lengths=new_lengths,
+            column_widths=new_widths,
+            dtypes=new_dtypes,
+        )
+
+    @classmethod
+    def _arrow_type_to_dtype(cls, arrow_type):
+        res = arrow_type.to_pandas_dtype()
+        if not isinstance(res, (np.dtype, str)):
```

Comment:
Why is this condition needed?

Reply:
Arrow function `to_pandas_dtype` may return both dtype objects and numpy types, this is to unify returned value type for simpler processing.

Reply:
I see, thanks!